server/src/main/scala/za/co/absa/atum/server/api/controller/CheckpointController.scala

```diff
@@ -17,31 +17,25 @@
 package za.co.absa.atum.server.api.controller
 
 import za.co.absa.atum.model.dto.CheckpointDTO
-import za.co.absa.atum.server.api.exception.ServiceError
 import za.co.absa.atum.server.api.service.CheckpointService
-import za.co.absa.atum.server.model.{ErrorResponse, GeneralErrorResponse, InternalServerErrorResponse}
+import za.co.absa.atum.server.model.ErrorResponse
 import zio._
 import zio.macros.accessible
 
 @accessible
-trait CheckpointController {
+trait CheckpointController extends BaseController {
```

Comment:
I don't think the service interface should provide implementation. `CheckpointControllerImpl` should extend `BaseController`.

Reply:
good point, I changed this in all services, controllers and repositories